src/java.base/share/classes/javax/security/auth/callback/NameCallback.java

```diff
@@ -42,7 +42,7 @@ public class NameCallback implements Callback, java.io.Serializable {
      * @serial
      * @since 1.4
      */
-    private String prompt;
+    private final String prompt;
```

Comment:
I think you can also mark `defaultName` final.

Reply:
First constructor doesn't set defaultName (or inputName), so there will be a error "might not have been initialized".

Reply:
I verified the error message happens when `defaultName` is final.

Reply:
Yes but `defaultName` (like `prompt`) should never change once the object is constructed. So it seems inconsistent not to also mark it final (and if necessary, set it to a default value (`null`) in the constructors).

Reply:
Agreed. I'll make the change.

Reply:
That works also, as the API allows for these values to be set/reset to null.